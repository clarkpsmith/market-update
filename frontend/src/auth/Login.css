.Login {
  text-align: left;
  background-color: black;
  height: 100vh;
  color: white;
  display: flex;
  justify-content: center;
}

.Login-form {
  text-align: left;
  padding: 25px;
  padding-left: 50px;

  width: 400px;

  background-color: black;
  border: 1px solid grey;
  border-radius: 2px;
}

.Login-input {
  border-radius: 3px;

  width: 90%;
  border: 1px solid rgb(216, 213, 213);
}

.Login-error {
  width: 90%;
  font-size: x-small;
  padding: 0px;
}

.Login-btn {
  margin: 25px;
  float: right;
  margin-right: 35px;
}

.Login-block {
  margin: 50px;
}

.Login-title {
  padding: 10px;
  text-align: center;
  color: rgb(200, 210, 236);
}
