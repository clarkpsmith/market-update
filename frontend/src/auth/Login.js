import React, { useState } from "react";
import "../common/SearchForm.css";
import Alert from "../helpers/Alert";
import { useHistory } from "react-router-dom";
import ChaseLoading from "../chaseloading/ChaseLoading";
import { Button, Form, FormGroup } from "reactstrap";
import "./Login.css";

const Login = ({ login }) => {
  const initialData = {
    username: "",
    password: "",
  };
  const [formData, setFormData] = useState(initialData);
  const [formErrors, setFormErrors] = useState(null);
  const [loading, setLoading] = useState(false);
  const history = useHistory();

  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData((data) => ({ ...data, [name]: value }));
  }

  const handleSubmit = async (evt) => {
    evt.preventDefault();
    setLoading(true);
    let result = await login(formData);

    if (result.success) {
      history.push("/watchlist");
    } else {
      setFormErrors(() => result.err);
      setLoading(false);
    }
  };
  if (loading) {
    return (
      <div className="Login">
        <ChaseLoading />
      </div>
    );
  }
  return (
    <div className="Login">
      <div className="Login-block">
        <h3 className="Login-title">Log In</h3>
        <Form className="Login-form" onSubmit={handleSubmit}>
          <FormGroup>
            <label htmlFor="username">Username</label>
            <br />
            <input
              className="Login-input"
              type="text"
              name="username"
              id="username"
              value={FormData.username}
              onChange={handleChange}
              required
            />
          </FormGroup>

          <FormGroup>
            <label htmlFor="password">Password</label>
            <br />
            <input
              className="Login-input"
              type="password"
              name="password"
              id="password"
              value={FormData.password}
              onChange={handleChange}
              required
            />
            <br />
          </FormGroup>
          {formErrors ? <Alert type="danger" messages={formErrors} /> : null}
          <Button size="md" className="Login-btn">
            Submit
          </Button>
          <br />
          <br />
        </Form>
      </div>
    </div>
  );
};
export default Login;
