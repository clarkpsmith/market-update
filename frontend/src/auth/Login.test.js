import React from "react";
import { render } from "@testing-library/react";
import Login from "./Login";

it("renders without crashing", () => {
  render(<Login />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Login />);
  expect(asFragment()).toMatchSnapshot();
});
