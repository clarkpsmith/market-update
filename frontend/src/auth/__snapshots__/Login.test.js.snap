// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <div
      class="Login-block"
    >
      <h3
        class="Login-title"
      >
        Log In
      </h3>
      <form
        class="Login-form"
      >
        <div
          class="form-group"
        >
          <label
            for="username"
          >
            Username
          </label>
          <br />
          <input
            class="Login-input"
            id="username"
            name="username"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="password"
          >
            Password
          </label>
          <br />
          <input
            class="Login-input"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
          <br />
        </div>
        <button
          class="Login-btn btn btn-secondary btn-md"
        >
          Submit
        </button>
        <br />
        <br />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
